app/models/spree/admin/main_menu/section.rb

```diff
@@ -0,0 +1,68 @@
+module Spree
+  module Admin
+    module MainMenu
+      class Section
+        attr_reader :key, :label_translation_key, :icon_key, :items
+
+        def initialize(key, label_translation_key, icon_key, availability_check, items)
+          @key = key
+          @label_translation_key = label_translation_key
+          @icon_key = icon_key
+          @availability_check = availability_check
+          @items = items
+        end
+
+        def add(item)
+          raise ArgumentError, "Item with key #{key} already exists" if index_for_key(item.key)
+
+          @items << item
+        end
+
+        def child_with_key?(key)
+          index_for_key(key).present?
+        end
+
+        def remove(item_key)
+          item_index = index_for_key(item_key)
+          raise ArgumentError, "Item not found for key #{item_key}" unless item_index
+
+          @items.delete_at(item_index)
+        end
+
+        def item_for_key(key)
+          @items.find { |e| e.key == key }
+        end
+
+        def insert_before(item_key, item_to_add)
+          item_index = index_for_key(item_key)
+          raise ArgumentError, "Item not found for key #{item_key}" unless item_index
```

Comment:
Maybe a `KeyError` would be more fitting here?

Reply:
Also - it seems that this logic (get index by key, raise if the key is nil) is used in multiple places. Maybe we could use a second function like `index_for_key!` or `fetch_index_for_key` (mirroring `Hash#fetch`) which would wrap the exception.

Reply:
You mean, adding a private method that would throw an exception if key is not found?